src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java

```diff
@@ -0,0 +1,638 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.controller.nsc;
+
+import static org.junit.Assert.*;
+import static org.osc.controller.nsc.SampleSdnRedirectionApiTestData.*;
+import static org.osc.sdk.controller.FailurePolicyType.NA;
+import static org.osc.sdk.controller.TagEncapsulationType.VLAN;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.EntityManager;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.osc.controller.nsc.api.SampleSdnRedirectionApi;
+import org.osc.controller.nsc.entities.InspectionHookEntity;
+import org.osc.controller.nsc.entities.InspectionPortEntity;
+import org.osc.controller.nsc.entities.NetworkElementEntity;
+import org.osc.controller.nsc.entities.PortGroupEntity;
+import org.osc.controller.nsc.utils.RedirectionApiUtils;
+import org.osc.sdk.controller.element.Element;
+import org.osc.sdk.controller.element.InspectionHookElement;
+import org.osc.sdk.controller.element.InspectionPortElement;
+import org.osc.sdk.controller.element.NetworkElement;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+public class SampleSdnRedirectionApiTest {
+
+    public EntityManager em;
+
+    @Mock(answer = Answers.CALLS_REAL_METHODS)
+    TestTransactionControl txControl;
+
+    @InjectMocks
+    private SampleSdnRedirectionApi redirApi;
+
+    @Before
+    public void setup() {
+        this.em = InMemDB.getEntityManager();
+        this.txControl.init(this.em);
+        setupDataObjects();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        InMemDB.close();
+    }
+
+    @Test
+    public void verifyCorrectNumberOfMacsAdPortIps() throws Exception {
```

Comment:
assume all of this code is moved from the previous integration test?

Any hot spots to look for?

Also is the plan to update them to the arrange, act and assert as part of next checkin?

Reply:
This was a simple copy, yes no actual tests were changed, except tearDown code and DB url.